app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,65 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!, only: [:shopping_list]
+
+  def index
+    @foods = current_user.foods.includes(:recipe_foods)
+  end
+
+  def show
+    @food = current_user.foods.find(params[:id])
+  end
```

Comment:
- [ ] [REQUIRED] As recommended by the previous reviewer, after successfully generating a shopping list there is a route error while trying to redirect to the shopping list index page when I click on the shopping list.

<img width="1112" alt="Screenshot 2023-11-25 at 21 50 09" src="https://github.com/hajnaloltyan/recipe-app/assets/58692590/d214049e-09eb-4404-990b-95c2eb9b106a">


<img width="965" alt="Screenshot 2023-11-25 at 21 34 18" src="https://github.com/hajnaloltyan/recipe-app/assets/58692590/28c73057-ed7c-4aa2-8f37-47855356c180">

- The error is being triggered when trying to find a Food with the ID "shopping_list. I can see that `params[:id]` is expected to be a numeric ID of the food you want to display. However, in the error message, params[:id] is set to "shopping_list," which is not a valid ID. For the show action to display properly, I resolved it with:
``` ruby
def show
  if params[:id] == "shopping_list"
    shopping_list
    render :shopping_list
  else
    @food = current_user.foods.find(params[:id])
  end
end
```

After I resolved it:
<img width="1086" alt="Screenshot 2023-11-25 at 21 51 08" src="https://github.com/hajnaloltyan/recipe-app/assets/58692590/e6f07b13-2383-4f1e-aaa4-df603df486a4">


Could we resolve this before we ask for another review?